src/routes/Organizations/Repository/Tabs/Tags/Filter.tsx

```diff
@@ -10,91 +10,165 @@ import {
     setFilter(value);
     setPagination({page: 1, perPage: 25});
   };
 
-  const kebabItems = [<DropdownItem key="delete">Delete</DropdownItem>];
+  const deleteTags = async () => {
+    await Promise.all(
+      selectedTags.map(async (tag) =>
+        deleteTag(props.organization, props.repository, tag),
+      ),
+    );
+    props.loadTags();
+    setIsModalOpen(!isModalOpen);
+    setSelectedTags([]);
+  };
+
+  const kebabItems = [
+    <DropdownItem
+      key="delete"
+      onClick={() => {
+        setIsModalOpen(!isModalOpen);
+      }}
+    >
+      Delete
+    </DropdownItem>,
+  ];
   const [isKebabOpen, setKebabOpen] = useState(false);
 
   return (
-    <Toolbar>
-      <ToolbarContent>
-        <ToolbarItem spacer={{default: 'spacerNone'}}>
-          {/* Need to hook up the functionality of this select */}
-          <Select
-            variant={SelectVariant.checkbox}
-            aria-label="Select Filter Input"
-            onToggle={() => setFilterOpen(!filterOpen)}
-            onSelect={() => {
-              console.log('selected');
+    <>
+      <Modal
+        title={`Delete the following tag${selectedTags.length > 1 ? 's' : ''}?`}
+        isOpen={isModalOpen}
+        disableFocusTrap={true}
+        data-testid="delete-tags-modal"
+        key="delete-tags-modal"
+        onClose={() => {
+          setIsModalOpen(!isModalOpen);
+        }}
+        variant={ModalVariant.small}
+        actions={[
+          <Button
+            key="cancel"
+            variant="primary"
+            onClick={() => {
+              setIsModalOpen(!isModalOpen);
             }}
-            selections={selected}
-            isOpen={filterOpen}
-            placeholderText="Filter"
           >
-            <SelectOption key={0} value="OS" />
-            <SelectOption key={1} value="Size" />
-          </Select>
-        </ToolbarItem>
-        <ToolbarItem>
-          <TextInput
-            isRequired
-            type="search"
-            id="modal-with-form-form-name"
-            name="search input"
-            placeholder="Filter by Tag name"
-            value={filter}
-            onChange={filterTags}
-          />
-        </ToolbarItem>
-        <ToolbarItem>
-          <Dropdown
-            onSelect={() => setKebabOpen(!isKebabOpen)}
-            toggle={
-              <KebabToggle
-                onToggle={() => setKebabOpen(!isKebabOpen)}
-                id="toggle-id-6"
+            Cancel
+          </Button>,
+          <Button key="delete" variant="primary" onClick={deleteTags}>
+            Delete
+          </Button>,
+        ]}
+      >
+        {selectedTags.map((tag) => (
+          <span key={tag}>
+            <Label>{tag}</Label>{' '}
+          </span>
+        ))}
+        {selectedTags.length > 1 ? (
+          <div>
+            <b>Note:</b> This operation can take several minutes.
+          </div>
+        ) : null}
+      </Modal>
+      <Toolbar>
```

Comment:
Is it possible to move the toolbar into another component?

Reply:
yes let me break this down a bit